// scss-lint/lib/utils.js
'use strict';

var EXIT_CODES = module.exports.EXIT_CODES = {
  ok:        0,
  warning:   1,  // One or more warnings (but no errors) were reported
  error:     2,  // One or more errors were reported
  usage:     64, // Command line usage error
  no_input:  66, // Input file did not exist or was not readable
  software:  70, // Internal software error
  config:    78, // Configuration error
}

var halt = module.exports.halt = function(type) {
  if (typeof type === 'undefined') {
    type = 'ok';
  }
  process.exit(EXIT_CODES[type]);
};

var requireAll = module.exports.requireAll = function(directory, exclude) {
  var items = {};
  exclude = typeof exclude !== 'undefined' ? exclude : '';
  require("fs").readdirSync(directory).forEach(function(item){
    if (item !== exclude) {
      var name = item.substr(0, item.lastIndexOf('.')) || item;
      items[name] = require(directory + '/' + item);
    }
  });
  return items;
};

var keywordToHex = module.exports.colorToHex = function(keyword) {
  return require('parse-color')(keyword).hex;
};

var isArray = module.exports.isArray = function(object) {
  return Object.prototype.toString.call(object) === '[object Array]';
};
